app/helpers/assessment_autograde_core.rb

```diff
@@ -412,13 +412,37 @@ def saveAutograde(submissions, feedback)
       @autograde_prop = @assessment.autograder
 
       # Record each of the scores extracted from the autoresult
-      scores.keys.each do |key|
-        problem = @assessment.problems.find_by(name: key)
-        raise AutogradeError.new("Problem \"" + key + "\" not found.") unless problem
-        submissions.each do |submission|
+      submissions.each do |submission|
+
+        # If modifySubmissionScore is overridden, use that to modify the score
+        # Provide it with the scores and previous submissions
+        if @assessment.overwrites_method?(:modifySubmissionScores)
+          
+          previous_submissions_lookback = if (@assessment.assessment_variable.key?("previous_submissions_lookback"))
+                                            @assessment.assessment_variable["previous_submissions_lookback"]
+                                          else
+                                            1000 # default to 1000
+                                          end 
+                                          
+          # Get previous submissions of the same assessment by the user, excluding the current submission
+          previous_submissions = Submission.where(course_user_datum_id: submission.course_user_datum_id, 
```

Comment:
Should we filter the submissions such that only submissions that have scores are included? Rn, if you create a submission that finishes grading before the submission before it, when you access the previous submission, it'll set it as 0 instead of using the score from the last-graded submission. We could also leave it up to the lab author to filter this out in their code though, but I guess it depends on whether you think there's any use-cases where lab authors want to use submissions that are still grading in progress.

![image](https://user-images.githubusercontent.com/50491000/229309645-0f67a9d6-9e49-46fb-a782-238d1f626734.png)

Reply:
I can add that as a boolean flag, `include_current_grading_submissions` to assessment_variable, defaulting to `false` how does that sound?